hare/certifiedblock.go

```diff
@@ -0,0 +1,121 @@
+package hare
+
+import (
+	"context"
+	"fmt"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/hare/config"
+	"github.com/spacemeshos/go-spacemesh/p2p/pubsub"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+type BlockCertificate struct {
+	blockID               types.BlockID
+	terminationSignatures []terminationSignature
+}
+
+func (bCert *BlockCertificate) IsValid() bool {
+	panic("not yet implemented")
+}
+
+// BlockCertifyingService allows a node to participate in the building of
+// certified blocks. This includes verifying & caching gossiped block signatures
+// and, if chosen by the oracle, participating in block-signing upon termination
+// of a given layer.
+type BlockCertifyingService struct {
+	rolacle Rolacle
+}
+
+// NewBlockCertifyingService constructs a new BlockCertifyingService.
+// It expects to receive the same config as Hare.
+func NewBlockCertifyingService(rolacle Rolacle, config config.Config) (*BlockCertifyingService, error) {
+
+	service := &BlockCertifyingService{}
+	service.rolacle = rolacle
+
+	return service, fmt.Errorf("not yet implemented")
+}
+
+func (s *BlockCertifyingService) Start(ctx context.Context) error {
+
+	return fmt.Errorf("not yet implemented")
+}
+
+func (s *BlockCertifyingService) GossipHandler() pubsub.GossipHandler {
```

Comment:
how is this going to be invoked? are you planning on creating a new func for each gossip msg? if so, why?

Reply:
yeah... but it is not consistent with other handlers. other handlers simply pass the func ptr directly. what is the advantage of you doing differently?